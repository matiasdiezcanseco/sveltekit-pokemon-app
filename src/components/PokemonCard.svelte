<script lang="ts">
	import { fade } from 'svelte/transition'
	export let pokeman: any
</script>

<a
	class="p-6 bg-gray-100 text-gray-800 text-center rounded-md shadow-sm hover:shadow-md flex flex-col items-center"
	href={`/pokemon/${pokeman.id}`}
	data-testid="card-testid"
	transition:fade
>
	<img class="w-40 h-40" src={pokeman.image} alt={pokeman.name} />
	<h2 class="uppercase text-2xl" data-testid="cardname-testid">{pokeman.id}. {pokeman.name}</h2>
</a>
