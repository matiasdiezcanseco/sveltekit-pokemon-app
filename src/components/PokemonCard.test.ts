import PokemonCard from './PokemonCard.svelte'
import { describe, test } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/svelte'

describe('PokemonCard', () => {
	const pokeman = {
		id: '150',
		name: 'Mewtwo',
		image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150'
	}

	test('should display id and name correctly ', () => {
		render(PokemonCard as any, {
			props: {
				pokeman
			}
		})
		expect(screen.getByTestId('cardname-testid').innerHTML).toEqual('150. Mewtwo')
	})

	test('should show image', () => {
		render(PokemonCard as any, {
			props: {
				pokeman
			}
		})
		expect(screen.getByRole('img')).toBeTruthy()
	})

	test('should match snapshot', () => {
		const wrapper = render(PokemonCard as any, {
			props: {
				pokeman
			}
		})
		expect(wrapper).toMatchSnapshot()
	})
})
