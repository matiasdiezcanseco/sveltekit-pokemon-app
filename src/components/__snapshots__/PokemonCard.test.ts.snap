// Vitest Snapshot v1

exports[`PokemonCard > should match snapshot 1`] = `
{
  "component": PokemonCard$ {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        {
          "id": "150",
          "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150",
          "name": "Mewtwo",
        },
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "pokeman": 0,
      },
      "root": <div>
        <a
          class="p-6 bg-gray-100 text-gray-800 text-center rounded-md shadow-sm hover:shadow-md flex flex-col items-center"
          data-testid="card-testid"
          href="/pokemon/150"
        >
          <img
            alt="Mewtwo"
            class="w-40 h-40"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150"
          />
           
          <h2
            class="uppercase text-2xl"
            data-testid="cardname-testid"
          >
            150
            . 
            Mewtwo
          </h2>
        </a>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <a
        class="p-6 bg-gray-100 text-gray-800 text-center rounded-md shadow-sm hover:shadow-md flex flex-col items-center"
        data-testid="card-testid"
        href="/pokemon/150"
      >
        <img
          alt="Mewtwo"
          class="w-40 h-40"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/150"
        />
         
        <h2
          class="uppercase text-2xl"
          data-testid="cardname-testid"
        >
          150
          . 
          Mewtwo
        </h2>
      </a>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
